### PR TITLE
Add Anchor IDs to Widget Context Areas.

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue #4288

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/googlesitekit/widgets/components/WidgetContextRenderer.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ import { Grid, Row, Cell } from '../../../material-components';
 const { useSelect } = Data;
 
 const WidgetContextRenderer = ( props ) => {
-	const { slug, className, Header, Footer } = props;
+	const { id, slug, className, Header, Footer } = props;
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/widgets/components/WidgetContextRenderer.js:36]: @kuasha420, please, add the new `id` prop to the component's propTypes list.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @kuasha420. This is almost looks good. Just one tiny request and it should be good to go. Please, take a look.

File: assets/js/googlesitekit/widgets/components/WidgetContextRenderer.js (hunk 1 above)
Comment (kuasha420) [reply assets/js/googlesitekit/widgets/components/WidgetContextRenderer.js:36]: @eugene-manuilov oops, coming from the TypeScript land, I keep forgetting this. Fixed!

**Review by eugene-manuilov — APPROVED** [review]
Awesome! Thanks, @kuasha420.

**Pull request merged**